docs/bin/dump_config.py

```diff
@@ -62,8 +63,19 @@ def main(args):
     output_name = os.path.join(output_dir, template_file.replace('.j2', ''))
     temp_vars = {'config_options': config_options}
 
-    with open(output_name, 'wb') as f:
-        f.write(template.render(temp_vars).encode('utf-8'))
+    data = template.render(temp_vars).encode('utf-8')
+
+    # Check if the destionation file is different, if not, return
+    write_out = True
+    if os.path.exists(output_name):
+        sha1_new = sha1(data).hexdigest()
+        sha1_old = sha1(open(output_name).read()).hexdigest()
+        if sha1_new == sha1_old:
```

Comment:
Why not simply compare contents, instead of computing sha1 of both first? Since you read both into memory, that's more efficient :)

Also: it currently fails under Python 3:
```
Traceback (most recent call last):
  File "../bin/dump_config.py", line 84, in <module>
    sys.exit(main(sys.argv[:]))
  File "../bin/dump_config.py", line 72, in main
    sha1_old = sha1(open(output_name).read()).hexdigest()
TypeError: Unicode-objects must be encoded before hashing
```